Features

phalado opened the pull request:
Project completed as Odin Project's requirement in https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations#project-2-private-events.

We did the "Setup and Sign in" where we created the model and controller for the Users.
In the "Basic events" we created the model and controller for creating, showing and listing events and did the association with the Users. Everything fine until here.
In the "Event Attendance" we created a model for attendance, that would be the link between users that attended in determined events and events that the users were invited for as well we modified the events to show the past and upcoming events in its show page.

phalado committed:
Redone until topic 5.

Commit:
created sessions controllers and view

Commit:
added event model

phalado committed:
Done until topic 3 of second section

phalado committed:
Uploading db files

Commit:
added event controller and views

phalado committed:
Did some associations

phalado committed:
Did some showing in users and events page

phalado committed:
Done with upcoming events and future events

Commit:
created scopes and views for past and upcoming events

Commit:
added root route

phalado committed:
Done with README and linter corrections

phalado committed:
Updated the rubocop.

Apparently LineLength is not from Metrics anymore, but Layouts.

wathigo commented on app/controllers/events_controller.rb:
```diff
@@ -0,0 +1,22 @@
+class EventsController < ApplicationController
+  def new
+    @event = Event.new
+  end
```
[CHANGES REQUIRED]
When I try to create a new event, I get this error.
![sknd](https://user-images.githubusercontent.com/44072711/72210782-a7985480-348e-11ea-9b5f-0086fd36af24.png)

wathigo reviewed (changes requested):
## Chages Required 
Hi guys :wave:  
There is an error preventing us from creating an event.
See the comment :point_up: 

**After fixing this issue, submit another request for review.**

**Happy Coding :computer: 
Simon Wathigo**

phalado committed:
Modifications done, now login works fine

Had to do some modifications in the login form, trading form_for for form_tag. Now you can, also, click on your name and go to your user's page.

maelfosso commented on app/controllers/events_controller.rb:
```diff
@@ -0,0 +1,22 @@
+
+  def show
+    @event = Event.find_by(id: params[:id])
+    @creator = User.find_by(id: @event.creator)
```
Not needed. You could directly access to the creator from `@event`

maelfosso commented on app/views/events/show.html.erb:
```diff
@@ -0,0 +1,17 @@
+  <%= @event.description%>
+  -->
+  <h2>Creator:</h2>
+  <%= @creator.name%>
```
Just do `@event.creator.name`. It will work!

maelfosso commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,34 @@
+      ev = Event.find_by(id: at.attended_event)
+      @upcoming_events << ev if ev.date >= DateTime.now
+      @prev_events << ev if ev.date < DateTime.now
+    end
```
You can simplify this part
```ruby
@user.attended_event.upcoming_events
@user.attended_event.prev_events
```

phalado replied:
Done

maelfosso commented on app/controllers/events_controller.rb:
```diff
@@ -0,0 +1,22 @@
+  end
+
+  def create
+    current_user = User.find_by(id: session[:user_id])
```
**`OPTIONAL`**
Define a global `current_user` variable in ApplicationController and set it when the user signed so that you will use it everywhere

phalado replied:
Done

maelfosso reviewed (changes requested):
Hello @phalado @wathigo  :wave:
 
Nice work. I really love you ui design.
About the work, there are some things to fix in the code and a main error happening when trying to create an event

-
![Screenshot from 2020-01-12 07-56-32](https://user-images.githubusercontent.com/8848104/72215368-97c25400-3512-11ea-9442-77c0e39f2d73.png)

I think it's because the `current_user` is nill which happens because the `session[:user_id]` has not been set. I got the error when trying to create an event after **creating an user**. I suggest to use [**Gem : `byebug`**](https://github.com/deivid-rodriguez/byebug) to debug your code for a better understanding.

- Also, you forgot to set a field to the date Event in the event creation form.

- Read the comment I left below, please, to improve your work.

Kindly make those changes and submit a new code review request.
Thanks

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

Commit:
fixed tsa suggestions

phalado committed:
Did rubocop corrections

samgaco commented on db/migrate/20200109163351_add_invitation_to_user.rb:
```diff
@@ -0,0 +1,5 @@
+class AddInvitationToUser < ActiveRecord::Migration[6.0]
+  def change
+    add_column :users, :invitation, :string
+  end
+end
```
- [x] You are adding a column to users but apparently the first migration setting up the table has been erased

samgaco reviewed (changes requested):
Hi @alexawesomecode and @phalado :high_brightness: 

There's a problem with your migrations, please fix that before continuing. 

Regards!

-----------------------

* ###### STATUS OF THE PROJECT: REQUESTED CHANGES :stop_sign:

phalado committed:
re-added some tables to db

phalado committed:
some last changes in the db's tables

samgaco commented:
You can move on guys :+1:
